src/msg/simple/Pipe.cc

```diff
@@ -332,7 +334,7 @@ int Pipe::accept()
     goto fail_unlocked;
   }
   {
-    bufferptr tp(sizeof(peer_addr));
+    bufferptr tp(sizeof(ceph_entity_addr));
```

Comment:
agreed

On Wed, May 11, 2016 at 5:43 PM, Kefu Chai <notifications@github.com> wrote:

> In src/msg/simple/Pipe.cc
> <https://github.com/ceph/ceph/pull/8974#discussion_r62819293>:
>
> > @@ -332,7 +334,7 @@ int Pipe::accept()
> >      goto fail_unlocked;
> >    }
> >    {
> > -    bufferptr tp(sizeof(peer_addr));
> > +    bufferptr tp(sizeof(ceph_entity_addr));
>
> i understand this is semanticly more correct.
>
> but can we make this explicit? say, we encode/decode this field with a
> entity_addr_t, while read a chunk sizing sizeof(ceph_entity_addr). maybe
> we can have a
>
> static_assert(sizeof(entity_addr_t) == sizeof(ceph_entity_addr))
>
> somewhere?
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/ceph/ceph/pull/8974/files/1f0893183f2829d1e22779d87b5146a7d2628593#r62819293>
>



-- 

Best Regards,

Wheat